utils/python/CIME/XML/grids.py

```diff
@@ -0,0 +1,99 @@
+"""
+Common interface to XML files which follow the grids format,
+This is not an abstract class - but inherits from the abstact class GenericXML
+"""
+
+from standard_module_setup import *
+from CIME.utils import expect, convert_to_string, convert_to_type
+from generic_xml import GenericXML
+
+logger = logging.getLogger(__name__)
+
+class Grids(GenericXML):
+
+    def __init__(self, infile=None):
+        GenericXML.__init__(self, infile)
+        self.groups={}
+
+    def print_values(self, verbose=None):
+        # write out help message
+        help_node = self.get_node("help")
+        helptext = help_node.text
+        print helptext
+
+        # if verbose mode is on, then also obtain nodes for domains and maps
+        if verbose is not None:
+            domain_nodes = self.get_nodes(nodename="domain")
+            gridmap_nodes = self.get_nodes(nodename="gridmap")
+            gridRE = re.compile(r"[_]{0,1}[a-z]{1,2}%")
+
+        # write out grid elements 
+        grid_nodes = self.get_nodes(nodename="grid")
+        for grid_node in grid_nodes:
+            lname   = grid_node.find("lname")
+            sname   = grid_node.find("sname")
+            support = grid_node.find("support")
+            alias   = grid_node.find("alias")
```

Comment:
@jedwards4b can correct me if I'm wrong, but I thought we wanted the logger to handle output and debug outout?